paypal/express/views.py

```diff
@@ -445,6 +486,8 @@ def render_to_response(self, methods, basket):
         else:
             # No shipping methods available - we flag this up to PayPal indicating that we
             # do not ship to the shipping address.
-            pairs.append(('NO_SHIPPING_OPTION_DETAILS', 1))
+            pass
+            # pairs.append(('NO_SHIPPING_OPTION_DETAILS', 1))
```

Comment:
Any particular reason to comment it out?

Reply:
This was an odd one actually, despite the pass the 'NO_SHIPPING_OPTION_DETAILS' was being set in every instance. I thought it might be an indenting issue, but no, so did this hack to resolve that. Will clean up and add comments.

Reply:
And the reason now discovered; per the python docs https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops

> Loop statements may have an else clause; it is executed when the loop terminates through exhaustion of the list (with for)

I think the original author probably coded the for: else: here expecting the else condition to only be reached when there were no methods set.My Python is pretty basic, so I'm not sure, and I'm not sure what the best way of resolving this is, I believe it should be something like

```
if methods:
    # set up pairs
else:
    # No shipping methods available - we flag this up to PayPal indicating that we
    # do not ship to the shipping address.
    pairs.append(('NO_SHIPPING_OPTION_DETAILS', 1))
```
so I'll refactor like that.